### ISSUE TITLE
Commandline options for parsing behavior

### ISSUE THREAD

**Issue by joakim-hove** [OPM/opm-simulators#1383]
I thought of this while reviewing: https://github.com/OPM/opm-parser/pull/1187

I think flow should have some command-line options to control the parsecontext strictness; i.e.

```
flow --language-lawyer
```
and
```
flow --totally-relaxed
```

**Comment by atgeirr** [comment]
I agree, I've thought of this but not implemented it yet. A few questions:
 - By default, everything is throwing, correct?
 - Is there a way to access a list of all possible options in code?
 - Will the environment variables always override everything? I think that such vars in general should override "program defaults" but not things set explicitly on the command line. I imagine this would be consistent with general UNIX behaviour, but I might be wrong.

**Comment by alfbr** [comment]
Having command line options is probably a good idea. However, please allow for people on the user side to determine what defaults to use.

**Comment by joakim-hove** [comment]
> By default, everything is throwing, correct?

Yes - that is correct.

> Is there a way to access a list of all possible options in code?

The `ParseContext` object supports iteration.

> Will the environment variables always override everything? 

The order of things is:

1. Default initialization - depending on which constructor you use that is either 'all throw' or if use [this constructor](https://github.com/OPM/opm-parser/blob/master/lib/eclipse/Parser/ParseContext.cpp#L49) which flow does you can initialize the keys individually.

2. Update according to environment variables.

3. Return the initialized `ParseContext` instance to calling scope.

After 3) `flow` will have the `ParseContext` instance and can update behaviour according to e.g. commandline options. So - I would it should be quite simple to eat your cake and have it too.
